test/test_transforms_v2_refactored.py

```diff
@@ -2904,6 +2956,55 @@ def test__get_params(self, sigma):
             assert sigma[0] <= params["sigma"][0] <= sigma[1]
             assert sigma[0] <= params["sigma"][1] <= sigma[1]
 
+    # np_img = np.arange(3 * 10 * 12, dtype="uint8").reshape((10, 12, 3))
+    # np_img2 = np.arange(26 * 28, dtype="uint8").reshape((26, 28))
+    # {
+    #     "10_12_3__3_3_0.8": cv2.GaussianBlur(np_img, ksize=(3, 3), sigmaX=0.8),
+    #     "10_12_3__3_3_0.5": cv2.GaussianBlur(np_img, ksize=(3, 3), sigmaX=0.5),
+    #     "10_12_3__3_5_0.8": cv2.GaussianBlur(np_img, ksize=(3, 5), sigmaX=0.8),
+    #     "10_12_3__3_5_0.5": cv2.GaussianBlur(np_img, ksize=(3, 5), sigmaX=0.5),
+    #     "26_28_1__23_23_1.7": cv2.GaussianBlur(np_img2, ksize=(23, 23), sigmaX=1.7),
+    # }
+    REFERENCE_GAUSSIAN_BLUR_IMAGE_RESULTS = torch.load(
+        Path(__file__).parent / "assets" / "gaussian_blur_opencv_results.pt"
+    )
+
+    @pytest.mark.parametrize(
+        ("canvas_size", "kernel_size", "sigma"),
+        [
+            ("small", (3, 3), 0.8),
+            ("small", (3, 3), 0.5),
+            ("small", (3, 5), 0.8),
+            ("small", (3, 5), 0.5),
+            ("large", (23, 23), 1.7),
+        ],
+    )
+    @pytest.mark.parametrize("dtype", [torch.float32, torch.float64, torch.float16])
+    @pytest.mark.parametrize("device", cpu_and_cuda())
+    def test_functional_image_correctness(self, kernel_size, sigma, canvas_size, dtype, device):
+        if dtype == torch.float16 and device == "cpu":
+            # skip float16 on CPU case
+            return
+
+        if canvas_size == "small":
+            data = torch.from_numpy(np.arange(3 * 10 * 12, dtype="uint8").reshape((10, 12, 3))).permute(2, 0, 1)
```

Comment:
This can probably be native pytorch, same below in the `else` clause

```suggestion
            data = torch.arange(3 * 10 * 12, dtype=torch.uint8).reshape((10, 12, 3)).permute(2, 0, 1)```